modules/wallet/transactions.go

```diff
@@ -86,6 +89,18 @@ func (w *Wallet) Transaction(txid types.TransactionID) (pt modules.ProcessedTran
 	return modules.ProcessedTransaction{}, false
 }
 
+// decodeProcessedTransaction decodes a marshalled processedTransaction
+func decodeProcessedTransaction(ptBytes []byte, pt *modules.ProcessedTransaction) bool {
+	err := encoding.Unmarshal(ptBytes, pt)
+	if err != nil {
+		// COMPATv1.2.1: try decoding into old transaction type
+		var oldpt v121ProcessedTransaction
+		err = encoding.Unmarshal(ptBytes, &oldpt)
+		*pt = convertProcessedTransaction(oldpt)
+	}
+	return err == nil
+}
```

Comment:
this should live in `database.go`, and should be called by the other `ProcessedTransaction` functions.

I'm also wondering if we should optimize the decoding a bit. We can get the `ConfirmationHeight` without decoding all the inputs and outputs, which is nice. But I'm not sure if it's going to be a bottleneck in practice. We'll see what the profiling suggests.